lib/modules/cpu.go

```diff
@@ -0,0 +1,82 @@
+package modules
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"github.com/davidscholberg/go-i3barjson"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// Cpu represents the configuration for the cpu block.
+type Cpu struct {
+	BlockConfigBase `yaml:",inline"`
+	Cpu             string  `yaml:"cpu"` // name of the cpu to display (from /proc/stat), empty = all
+	CritUsage       float64 `yaml:"crit_usage"`
+
+	lastIdle  uint64
+	lastTotal uint64
```

Comment:
I'm a little bit uneasy with putting the `lastIdle` and `lastTotal` variables in this struct. Strictly speaking, this struct is for configuration data, not dynamic data that needs to persist over update calls. However, I do realize that there's currently no good way to persist dynamic data from one update call to the next.

I'm kind of leaning towards making these global variables rather than fields of the Cpu struct. I know that's not ideal, but it will at least keep dynamic data out of the configuration struct until we have a better solution for persisting data.

Any thoughts on this?